homeWork6/lesson1/Program.cs

```diff
@@ -0,0 +1,217 @@
+    Console.WriteLine($"Аккаунт \"{ account1.AccountNumber }\": { Math.Round(account1.Balance, 2) }\n");
+}
+
+static string? GetReverseString(char[] array) 
+{
+    //вернуть обратную строку (тип строка)
+    char tmp;
+    for(int i = 0, j = array.Length - 1; i < array.Length / 2; i++, j--)
+    {
+        tmp = array[i];
+        array[i] = array[j];
+        array[j] = tmp;
+    }
+    return new string(array);
+}
+public class BankAccount
+{
+    public BankAccount()
+    {
+        _counter += 1;//AccountNumber = 0;//"";
+        _accountNumber = _counter;
+    }
+
+    //accountNumber
+    private uint _accountNumber;// = 0;
+    public uint AccountNumber
+    {
+        get => _accountNumber;
+        set =>_accountNumber = value;//++_counter;//_accountNumber = String.Format("{0:00000}", ++_counter);
+    }
+
+    //accountType
+    public enum AccountType { Current, Estimated, Credit, Deposit };
+    private AccountType _accountType;
+    public AccountType CurrentAccountType
+    {
+        get => _accountType;
+        set => _accountType = value;
+    }
+
+    //balance
+    private double _balance;
+    public double Balance
+    {
+        get => _balance;
+        set => _balance = value;
+    }
+
+    private static uint _counter = 0;
+    private static uint Counter
+    {
+        get => _counter;
+        //set => _counter = value;
+    }
+
+    public bool GetMoney(double value)
+    {
+        //снятие со счета
+        double balance = _balance;
+        //
+        if (balance - value >= 0)
+        {
+            //деньги можно снять, изменяем баланс
+            _balance -= value;
+            return true;
+        }
+        return false;
+    }
+
+    public bool SetMoney(double value)
+    {
+        //положить на счет
+        if (value > 0)
+        {
+            _balance += value;
+            return true;
+        }
+        return false;
+    }
+
+    public bool TransferMoney(BankAccount bankAccount, double amount)//double value
+    {
+        //перевести деньги на счет,
+        //bankAccount - объект, у которого снимаются деньги
+        if (bankAccount != null)
+        {
+            if (amount > 0 && bankAccount.Balance - amount >= 0)
+            {
+                //на счете есть нужная сумма
+                bankAccount.Balance -= amount;
+                this.SetMoney(amount);              
+            }
+            return true;
+        }
+        return false;
+    }
+
+    public static bool operator ==(BankAccount account1, BankAccount account2)
+    {
+        if (ReferenceEquals(account1, account2)) 
+        {
+            return true;
+        }
+            
+        if (ReferenceEquals(account1, null)) 
+        {
+            return false;
+        }
+            
+        if (ReferenceEquals(account2, null))
+        {
+            return false;
+        }
+        return account1.Equals(account2);

```

Comment:
Оператор == реализуется через вызов метода Equals
```suggestion
        return Equals(account1, account2);
```